File: Output/java/medium/LongestSubstringWithoutRepeatingCharacters.java
/**https://leetcode.com/problems/longest-substring-without-repeating-characters/ */
//Given a string s, find the length of the longest substring without repeating characters.


class LongestSubstringWithoutRepeatingCharacters {
    public int lengthOfLongestSubstring(String s) {
        int[] map = new int[128];
        int left = 0;
        int right = 0;
        int max = 0;
        while(right < s.length()){
            char c = s.charAt(right);
            map[c]++;
            while(map[c] > 1){
                map[s.charAt(left)]--;
                left++;
            }
            max = Math.max(max, right - left + 1);
            right++;
        }
        return max;
        
    }
}

    

File: Tool/PreUtil/src/output/javascript/medium/LongestSubstringWithoutRepeatingCharacters.js
/**https://leetcode.com/problems/longest-substring-without-repeating-characters/ */
//Given a string s, find the length of the longest substring without repeating characters.


/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let map = {};
    let max = 0;
    let start = 0;
    for(let i = 0; i < s.length; i++){
        let char = s[i];
        if(map[char] != undefined){
            start = Math.max(start, map[char] + 1);
        }
        map[char] = i;
        max = Math.max(max, i - start + 1);
    }
    return max;
};

    